tests/test_client.py

```diff
@@ -0,0 +1,109 @@
+"""Tests of low-level client construction."""
+
+import json
+
+import pytest
+
+import tiledb.cloud.client
+from tiledb.cloud._common.api_v4.api import assets_api
+
+
+class NoResponseError(Exception):
+    """Raise when an intercepted request has no response."""
+
+
+@pytest.fixture(scope="function")
+def workspace_header_check(monkeypatch):
+    """Intercept server requests and checks the headers."""
+
+    def fake_and_check_request(self, method, url, *args, **kwargs):
+        assert kwargs["headers"]["X-TILEDB-WORKSPACE-ID"] == "workspace"
+        raise NoResponseError("Intercepted request has no response")
+
+    monkeypatch.setattr("urllib3.PoolManager.urlopen", fake_and_check_request)
```

Comment:
The gist of this is we patch the object that makes the connection to our server and have the replacement assert the existence of the workspace id header. No request is made.